server/api/impl/upstream/uphttp/upstream.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2020-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package uphttp
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+
+	"github.com/mattermost/mattermost-plugin-apps/server/api"
+	"github.com/mattermost/mattermost-plugin-apps/server/api/impl/upstream"
+	"github.com/mattermost/mattermost-plugin-apps/server/utils/httputils"
+)
+
+type Upstream struct {
+	rootURL   string
+	appSecret string
+}
+
+func NewUpstream(app *api.App) *Upstream {
+	return &Upstream{app.Manifest.HTTPRootURL, app.Secret}
+}
+
+func (u *Upstream) InvokeNotification(n *api.Notification) error {
+	// TODO
+	resp, err := u.post("", u.rootURL+"/notify/"+string(n.Subject), n)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	return nil
+}
+
+func (u *Upstream) InvokeCall(call *api.Call) *api.CallResponse {
+	resp, err := u.post(call.Context.ActingUserID, u.rootURL+call.URL, call)
```

Comment:
Maybe check if `call.URL` begins with a slash? If we want to do that, we can compute this full call URL in a helper function.

Reply:
1/5 call "URLs" (paths) must start with a `/` semantically; if they don't - the resulting URL will be invalid.

0/5 consider renaming `Call.URL` to `Call.Path`